src/shady-css/token-cursor.js

```diff
@@ -0,0 +1,36 @@
+const index = Symbol('index');
+const lexer = Symbol('lexer');
+const nextToken = Symbol('nextToken');
+
+class TokenCursor {
```

Comment:
what's the point of TokenCursor? Lexer is already stateful, with `offset`. Parser should be able to call directly into Lexer with `_lexer.nextToken()`.